docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,450 @@
+  valid range before density is computed, meaning that contamination with
+  invalid values could go unnoticed.  Still, it does appear that
+  ``displacedDensity`` is currently only used starting at index ``k = 2``, so
+  no invalid values should be getting used.
+
+Here is a (by no means complete) checklist of routines not used by
+``sub_ice_shelf_2D`` test cases that are a lower priority to update:
+
+Init:
+
+* [ ] ``ocn_init_cell_markers``:
+
+  * [ ] ``ocn_mark_maxlevelcell()``
+
+* [ ] ``ocn_init_global_ocean``:
+
+  * [ ] ``ocn_init_setup_global_ocean()``
+
+  * [ ] ``ocn_init_setup_global_ocean_create_model_topo()``
+
+  * [ ] ``ocn_init_setup_global_ocean_deepen_critical_passages()``
+
+  * [ ] ``ocn_init_setup_global_ocean_interpolate_land_ice_topography()``
+
+  * [ ] ``ocn_init_setup_global_ocean_modify_temp_under_land_ice()``
+
+  * [ ] ``ocn_init_setup_global_ocean_cull_inland_seas()``
+
+  * [ ] ``ocn_init_setup_global_ocean_interpolate_tracers()``
+
+* [ ] ``ocn_init_isomip``:
+
+  * [ ] ``ocn_init_setup_isomip()``
+
+* [ ] ``ocn_init_isomip_plus``:
+
+  * [ ] ``ocn_init_setup_isomip_plus()``
+
+Forward:
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``ocn_filter_btr_mode_tend_vel()``
+
+* [ ] ``ocn_gm``:
+
+  * [ ] ``ocn_GM_compute_Bolus_velocity()``
+
+...
+
+
+Implementation: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+The subroutines  ``ocn_thick_surface_flux_tend()`` and
+``ocn_tracer_surface_flux_tend()`` already distribute surface fluxes over
+multiple vertical levels in a loop, so surface fluxes will simply require the
+same treatment as any other loops.
+
+It will be a little trickier to make sure we perform proper indexing of all
+3D variables to get their "surface" values.  Here, the plan is to make sure
+that variables are set to the NetCDF fill value (a large, negative number)
+when they are invalid so that contamination should be obvious.
+
+Here is a checklist of variables or subroutines requiring special care because
+of top indexing that might not be easy to find (e.g. by searching for
+``maxLevel``):
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``tracersSurfaceValue``
+
+  * [ ] ``normalVelocitySurfaceLayer``
+
+  * [ ] ``ssh``
+
+  * [ ] ``fracAbsorbed``
+
+  * [ ] ``fracAbsorbedRunoff``
+
+  * [ ] ``nonLocalSurfaceTracerFlux``
+
+  * [ ] ``surfaceBuoyancyForcing``
+
+  * [ ] ``topDrag``
+
+  * [ ] ``topDragMag``
+
+  * [ ] ``landIceFrictionVelocity``
+
+...
+
+Implementation: Inactivate cells during simulation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+The main considerations here will be:
+
+* How time-consuming will it be to call
+  ``ocn_init_routines_compute_min_max_level()`` each time ``minLevelCell`` has
+  changed?  Is there anything we want to do now to make sure it is efficient?
```

Comment:
I can't think of any changes for the Phase 1 stage. I imagine for Phase 2 we could designate block(s) of cells that needs updating (likely in a separate regrid routine).

Reply:
A agree.  This subroutine doesn't look as heavy as I was afraid.  But we might still hesitate to run it every time step.  I just don't want us to lose track of this long-term need.